src/utils/searchViewHelper.js

```diff
@@ -50,6 +53,22 @@ const searchViewHelper = {
       searchViewHelper.searchByRevisionOrEmail(repository, search);
     }
   },
+
+  handleClickOutsideInput(event) {
+    try {
+      if (
+        event.target.id === 'search-revision-input' ||
+        event.target.classList.contains('search-revision') ||
+        event.target.parentElement.classList.contains('search-revision')
+      ) {
```

Comment:
I realized it was not as simple as detecting if the input was clicked or not. If the results were clicked, they would disappear as they are not the input or encapsulated by it.

there may be some edge cases not covered but this was how I was able to get it to behave as expected.

Reply:
Another way that could work is to add 3 event handlers:
1. on the document
2. twice the same handler on both the text input and the result list, that would call `e.stopPropagation()`. Then the event wouldn't bubble to the root element and event handler 1 wouldn't run.

Advantages: you don't need to hardcode ids or classnames.

[This article](https://www.freecodecamp.org/news/event-propagation-event-bubbling-event-catching-beginners-guide/#how-event-bubbling-happens-in-react) seems to give a very good explanation about all this event bubbling thing.


If you want to keep it this way, you may find https://developer.mozilla.org/en-US/docs/Web/API/Element/matches interesting as it could possibly simplify your if condition: `if (event.target.matches("#search-revision-input, .search-revision, .search-revision *")`. I'm not so sure this is better though :-)

Reply:
#89